src/components/ArticlePage/index.jsx

```diff
@@ -9,32 +9,50 @@ import RatingStarsBox from '../../components/RatingStarsBox';
 import HorizontalLine from '../../components/HorizontalLine';
 import TextAreaInput from '../../components/TextAreaInput';
 import CommentEntries from '../../components/CommentEntries';
-import { getOneArticle } from '../../redux/actions/articles';
+import { getOneArticle, rateArticle, getArticleAvgRating } from '../../redux/actions/articles';
+import { toast } from 'react-toastify';
 import './style.scss';
 
 
 export class SingleArticleView extends Component {
 
-  componentDidMount() {
-    const { fetchArticle, match } = this.props;
+  async componentDidMount() {
+    const { fetchArticle, getArticleAvgRating, match } = this.props;
     let articleId = match.params.articleId;
-    fetchArticle(articleId);
+    await fetchArticle(articleId);
+    getArticleAvgRating(articleId)
+  }
+
+  starClickHandle(item) {
+    let articleId = this.props.match.params.articleId;
+    const _this = this;
+    this.props.rateArticle(item, articleId )
+    .then(() => toast.success('Thank you for rating this article'))
+    .catch(function errorHandler(error) {
+      if (error.response.data.message === 'You already rated this article'){
+        return toast.error('You have already rated this article');
+      } else if(error.response.data === 'Unauthorized'){
+         toast.error('You need to login to rate an article');
+        return  _this.props.history.push('/login')
+      }
+      toast.error('Error while rating article make sure you are logged in');
+    })
   }
 
   render() {
     const { match, articles = [] } = this.props;
```

Comment:
@Princesso Consider doing this on `line 43` `const { match: { params: { articleId }, articles = [] } = this.props;`. That way there won't be a need for `line 44`.

Reply:
@ifiokeyo. Usually, functions that are declared with the function keyword have a different scope. The convention is to use named functions everywhere so in the error handler, the function has a different scope from its parent scope. 

Using unnamed functions however does not change the scope so the this keyword can work with unnamed functions unlike named functions.

Reply:
I will effect this change and update this PR, thanks.